internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -530,19 +530,19 @@ func (r *Reconciler) reconcileCluster(ctx context.Context, s *scope.Scope) error
 	} else {
 		log.Info("Patching Cluster", "diff", string(changes))
 	}
-	if err := patchHelper.Patch(ctx); err != nil {
+	modifiedResourceVersion, err := patchHelper.Patch(ctx)
+	if err != nil {
 		return errors.Wrapf(err, "failed to patch Cluster %s", klog.KObj(s.Current.Cluster))
 	}
 	r.recorder.Eventf(s.Current.Cluster, corev1.EventTypeNormal, updateEventReason, "Updated Cluster %q", klog.KObj(s.Current.Cluster))
 
 	// Wait until Cluster is updated in the cache.
 	// Note: We have to do this because otherwise using a cached client in the Reconcile func could
 	// return a stale state of the Cluster we just patched (because the cache might be stale).
-	// Note: It is good enough to check that the resource version changed. Other controllers might have updated the
-	// Cluster as well, but the combination of the patch call above without a conflict and a changed resource
-	// version here guarantees that we see the changes of our own update.
 	// Note: Using DeepCopy to not modify s.Current.Cluster as it's not trivial to figure out what impact that would have.
-	return clientutil.WaitForCacheToBeUpToDate(ctx, r.Client, "Cluster update", s.Current.Cluster.DeepCopy())
+	cluster := s.Current.Cluster.DeepCopy()
+	cluster.ResourceVersion = modifiedResourceVersion
+	return clientutil.WaitForCacheToBeUpToDate(ctx, r.Client, "Cluster update", cluster)
```

Comment:
This wait was a no-op previously because patchHelper.Patch did not update s.Current.Cluster during the Patch call so it always had the resourceVersion that we got at the beginning of the reconcile from the apiserver